apollo-integration/src/test/java/com/apollographql/apollo/ApolloLiveDataTest.kt

```diff
@@ -0,0 +1,182 @@
+import android.arch.lifecycle.Observer
+import com.apollographql.apollo.Utils.*
+import com.apollographql.apollo.api.Input
+import com.apollographql.apollo.cache.normalized.lru.EvictionPolicy
+import com.apollographql.apollo.cache.normalized.lru.LruNormalizedCacheFactory
+import com.apollographql.apollo.fetcher.ApolloResponseFetchers.NETWORK_ONLY
+import com.apollographql.apollo.integration.normalizer.EpisodeHeroNameQuery
+import com.apollographql.apollo.integration.normalizer.HeroAndFriendsNamesWithIDsQuery
+import com.apollographql.apollo.integration.normalizer.type.Episode
+import com.apollographql.apollo.livedata.ApolloLiveDataResponse
+import com.apollographql.apollo.livedata.toLiveData
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.verifyNoMoreInteractions
+import okhttp3.Dispatcher
+import okhttp3.OkHttpClient
+import okhttp3.mockwebserver.MockWebServer
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.CoreMatchers.instanceOf
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.After
+import org.junit.Assert.assertNotNull
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import java.io.IOException
+
+@RunWith(JUnit4::class)
+class ApolloLiveDataTest {
+    private lateinit var apolloClient: ApolloClient
+    private lateinit var mockWebServer: MockWebServer
+
+    private val FILE_EPISODE_HERO_NAME_WITH_ID = "EpisodeHeroNameResponseWithId.json"
+    private val FILE_EPISODE_HERO_NAME_CHANGE = "EpisodeHeroNameResponseNameChange.json"
+    private val FILE_HERO_AND_FRIEND_NAMD_CHANGE = "HeroAndFriendsNameWithIdsNameChange.json"
+
+    @Rule
+    @JvmField
+    val instantExecutorRule = InstantTaskExecutorRule()
+
+    @Before
+    @Throws(IOException::class)
+    fun setup() {
+        mockWebServer = MockWebServer()
+        mockWebServer.start()
+
+        val okHttpClient = OkHttpClient.Builder()
+                .dispatcher(Dispatcher(immediateExecutorService()))
+                .build()
+
+        apolloClient = ApolloClient.builder()
+                .serverUrl(mockWebServer.url("/"))
+                .dispatcher(immediateExecutor())
+                .okHttpClient(okHttpClient)
+                .normalizedCache(LruNormalizedCacheFactory(EvictionPolicy.NO_EVICTION), IdFieldCacheKeyResolver())
+                .build()
+    }
+
+    @After
+    @Throws(IOException::class)
+    fun stopServer() {
+        mockWebServer.shutdown()
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun callProducesValue() {
+        mockWebServer.enqueue(mockResponse(FILE_EPISODE_HERO_NAME_WITH_ID))
+
+        val observer = mock<Observer<ApolloLiveDataResponse<EpisodeHeroNameQuery.Data>>>()
+        val liveData = apolloClient.query(EpisodeHeroNameQuery(Input.fromNullable(Episode.EMPIRE))).toLiveData()
+        liveData.observeForever(observer)
```

Comment:
Have you considered LiveDataTestUtil?
https://github.com/googlesamples/android-architecture-components/blob/master/GithubBrowserSample/app/src/test-common/java/com/android/example/github/util/LiveDataTestUtil.kt

Reply:
Hi, I added `LiveDataTestUtil` on Apollo-integration module, 
and it used in `ApolloLiveDataTest` class.